env0/resource_environment.go

```diff
@@ -179,10 +179,11 @@ func resourceEnvironment() *schema.Resource {
 				Computed:    true,
 			},
 			"revision": {
-				Type:        schema.TypeString,
-				Description: "the revision the environment is to be run against",
-				Optional:    true,
-				Computed:    true,
+				Type:          schema.TypeString,
+				Description:   "the revision the environment is to be run against",
+				Optional:      true,
+				Computed:      true,
+				ConflictsWith: []string{"without_template_settings.0.revision"},
```

Comment:
I think we should conflict with `without_template_settings` in general.

If `without_template_settings` block **exists**, the revision shouldn't exist on the resource directly

If there's a `without_template_settings` but no `without_template_settings.revision`, and there's a `revision: master`, the revision used will be `''`

Reply:
df5fe0543593a5e9dc6b0d8b563341cfbae48dbf